tests/test_service_discovery.py

```diff
@@ -0,0 +1,35 @@
+"""Test common rest operations wrapper.
+"""
+import os
+import unittest
+
+import requests_mock
+
+from pyms.constants import CONFIGMAP_FILE_ENVIRONMENT
+from pyms.flask.app import Microservice
+from pyms.flask.services.service_discovery import ServiceDiscoveryConsul
+
+
+class ServiceDiscoveryTests(unittest.TestCase):
+    """Test common rest operations wrapper."""
+
+    BASE_DIR = os.path.dirname(os.path.abspath(__file__))
+
+    def setUp(self):
+        os.environ[CONFIGMAP_FILE_ENVIRONMENT] = os.path.join(self.BASE_DIR, "config-tests-service-discovery.yml")
+        ms = Microservice(path=__file__)
+        self.ms = ms
+
+    @requests_mock.Mocker()
+    def test_init(self, mock_request):
+        url = "http://localhost:8500/v1/agent/check/register"
+
+        mock_request.put(url)
+        self.ms.create_app()
+
+        self.assertTrue(True)
```

Comment:
What exactly is supposed to be tested here?

Reply:
If the method `self.ms.create_app()` not request the `url` with PUT Method  `mock_request raise` a assertionError

Reply:
Considerinng that external library is used here, using correct API endpoint should be already tested there (in `consulate`). So if anything proper `consulate` instantiation should be tested. Anyway this assertion itself literally does nothing.

Reply:
This assertion itself does nothing. Also from my experience there is no point in testing that external library (`consulate`) works as expected (i.e. calls correct endpoint on init) as it should already have it covered by tests. Instead correct call to `register_service` should be checked.